pennylane/transforms/hamiltonian_expand.py

```diff
@@ -67,14 +68,8 @@ def hamiltonian_expand(tape, group=True):
     >>> fn(res)
     -0.5
 
-    .. Warning::
-
-         Note that defining Hamiltonians inside of QNodes using arithmetic can lead to errors.
-         See :class:`~pennylane.Hamiltonian` for more information.
```

Comment:
I think this warning is unnecessary with the previous removal of the error for observables without owners.